### PR TITLE
Adapts billing history to invoice grain

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [#7]

**This PR will result in the following new package version:** `v0.2.0`

It's a breaking change as a number of fields were removed or created in `zuora_billing_history`. 

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
- Redesigned `zuora__billing_history` and `int_zuora__billing_enriched` so `invoice_id` was the actual grain, accounting for the cases when invoices can be tied to multiple credit balance adjustments and payments. This required the following steps:
    - All metrics associated with `credit_balance_adjustment_id` were aggregated on the invoice level (number of credit balance adjustments,total credit balance amount in home currency, first and last dates of credit balance adjustments on the invoice). Fields associated with an individual `credit_balance_adjustment_id` were removed.
    - All metrics associated with `payment_id` were aggregated on the invoice level (number of payments, total payment amount in home currency, first and last dates of payments on the invoice). Fields associated with an individual `payment_id` were removed.
    - Aggregated count of `payment_method_id` to gather a count of payment methods on each invoice.
- Updated yml of `zuora__billing_history` with aggregated fields, and removed non-aggregated fields associated with credit balance adjustments, payments, and payment methods.
- Modified `int_zuora__account_enriched` to account for new aggregated metric fields being pulled from `zuora__billing_history`. 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

Made sure that the models ran on `dbt run --vars`, particularly for adjusting the credit_balance_adjustment and taxation_item variables which are part of the CTEs that are modified.

For further validation, I recommend using the updated seed files and testing against the `main` branch to reproduce the bug, then making sure the new branch passes all tests. 

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [NA] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🪙

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+    - All metrics associated with `credit_balance_adjustment_id` were aggregated on the invoice level (number of credit balance adjustments,total credit balance amount in home currency, first and last dates of credit balance adjustments on the invoice). Fields associated with an individual `credit_balance_adjustment_id` were removed.
+    - All metrics associated with `payment_id` were aggregated on the invoice level (number of payments, total payment amount in home currency, first and last dates of payments on the invoice). Fields associated with an individual `payment_id` were removed.
+    - Aggregated count of `payment_method_id` to gather a count of payment methods on each invoice.
+- Updated yml of `zuora__billing_history` with aggregated fields, and removed non-aggregated fields associated with credit balance adjustments, payments, and payment methods.
+- Modified `int_zuora__account_enriched` to account for new aggregated metric fields being pulled from `zuora__billing_history`. 
+
+## 🔧 Under The Hood 🔩
+- Updated seed files in `integration_tests` to reproduce the initial `invoice_id` test issue for multiple credit balance adjustments and payments and test resolution.
```
Comment (fivetran-reneeli) [inline CHANGELOG.md:13]: I would just specify here that it was a uniqueness test issue, maybe 

"...uniqueness test issue for multiple credit balance adjustments and payments for one invoice".

File: models/intermediate/int_zuora__account_enriched.sql [hunk 2]
```diff
@@ -61,20 +61,20 @@ account_totals as (
 
     select 
         account_id,
+        min(most_recent_payment_date) as most_recent_payment_date,
```
Comment (fivetran-reneeli) [inline models/intermediate/int_zuora__account_enriched.sql:64]: Shouldn't this be max?

File: models/intermediate/int_zuora__account_enriched.sql [hunk 3]
```diff
@@ -61,20 +61,20 @@ account_totals as (
...
 
-        {% set sum_cols = ['refund_amount', 'discount_charges', 'tax_amount', 'invoice_amount', 'invoice_amount_home_currency', 'invoice_amount_paid', 'invoice_amount_unpaid'] %}
-        {% for col in sum_cols %}
-            sum({{ col }}) as total_{{ col }},
-        {% endfor %}
+        {% if var('zuora__using_credit_balance_adjustment', true) %}
+        min(most_recent_credit_balance_adjustment_date) as most_recent_credit_balance_adjustment_date,
```
Comment (fivetran-reneeli) [inline models/intermediate/int_zuora__account_enriched.sql:67]: same here, wouldn't most recent be max?

Comment (fivetran-reneeli) [reply models/intermediate/int_zuora__account_enriched.sql:67]: seems to be max based on the original logic too

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
...
+    - All metrics associated with `credit_balance_adjustment_id` were aggregated on the invoice level (number of credit balance adjustments,total credit balance amount in home currency, first and last dates of credit balance adjustments on the invoice). Fields associated with an individual `credit_balance_adjustment_id` were removed.
+    - All metrics associated with `payment_id` were aggregated on the invoice level (number of payments, total payment amount in home currency, first and last dates of payments on the invoice). Fields associated with an individual `payment_id` were removed.
+    - Aggregated count of `payment_method_id` to gather a count of payment methods on each invoice.
+- Updated yml of `zuora__billing_history` with aggregated fields, and removed non-aggregated fields associated with credit balance adjustments, payments, and payment methods.
+- Modified `int_zuora__account_enriched` to account for new aggregated metric fields being pulled from `zuora__billing_history`.
```
Comment (fivetran-reneeli) [inline CHANGELOG.md:10]: I would also add a bullet for removing the payment and credit balance fields in `int_zuora__billing_enriched` too

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated!

File: models/intermediate/int_zuora__account_enriched.sql (hunk 2 above)
Comment (fivetran-avinash) [reply models/intermediate/int_zuora__account_enriched.sql]: Good find, fixed!

File: models/intermediate/int_zuora__account_enriched.sql (hunk 3 above)
Comment (fivetran-avinash) [reply models/intermediate/int_zuora__account_enriched.sql]: Ugh, yes. Thanks for spotting.

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-avinash) [reply CHANGELOG.md:10]: I updated the syntax in Lines 6/7 to account for that!

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-reneeli Should be good for re-review!

**Review by fivetran-reneeli — APPROVED** [review]
Looks good, just remember to change the deps!

**Pull request merged**